### PR TITLE
Bhr 23

### PR DESCRIPTION
**Pull request by RobertHebel**

### CODE REVIEW

File: backend/src/user/router.spec.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ describe('User router', () => {
         .expect(HTTPStatus.OK)
         .then(res => picker(res.body));
 
+      console.log(user);
```
Comment (adam-golab) [inline backend/src/user/router.spec.js]: Unnecessary `console.log`.

File: frontend/src/reducers/authReducer.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      // Maybe change reduxPromise to accept request/success/failure
+      // as functions instead of only types.
+      throw new SubmissionError({ _error: payload.error.message });
+    default:
+      return state;
+  }
+};
+
+// Change that to change the expiration
+export const isAuthenticated = state => state.token && state.data.sub && true || false;
```
Comment (adam-golab) [inline frontend/src/reducers/authReducer.js]: What with the situation when user refresh a page?

File: frontend/src/actions/authActions.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      email: getState().auth.data.email,
+    };
+
+    return client.post(`/retrieve-password/${recoveryToken}`, { data: postData });
+  },
+});
+
+export const logoutAction = () => ({ type: AUTH_LOGOUT });
+
+export const redirectAfterLoginAction = () => delayedThunkCreator(1000, push('/welcome'));
```
Comment (adam-golab) [inline frontend/src/actions/authActions.js]: What is this timeout for?

File: frontend/src/reducers/authReducer.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      return forgotPwdRequestReducer(state, payload);
+    case FORGOT_PWD_SUCCESS:
+      return forgotPwdSuccessReducer(state, payload.result.body.token);
+    case LOGIN_FAILURE:
+    case CHANGE_PWD_FAILURE:
+    case FORGOT_PWD_FAILURE:
+      // There should be a better way....
+      // Maybe change reduxPromise to accept request/success/failure
+      // as functions instead of only types.
+      throw new SubmissionError({ _error: payload.error.message });
```
Comment (adam-golab) [inline frontend/src/reducers/authReducer.js]: Why do you throw an error here?

File: backend/src/index.js [hunk 5]
```diff
@@ -1,17 +1,28 @@
...
+
 app.get('/', function(req, res) {
   res.send('Hello World!');
 });
 
+app.use('/', authRouter);
 app.use('/user', userRouter);
 
 db.sequelize.sync()
+  .then(() => db.models.User.create({
```
Comment (adam-golab) [inline backend/src/index.js]: This code adds user every time that you start API.

File: frontend/src/actions/thunkActions.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export const delayedThunkCreator = (time, action) => dispatch =>
```
Comment (adam-golab) [inline frontend/src/actions/thunkActions.js]: This timeout should be on the backend side

File: frontend/src/services/apiClient.js [hunk 7]
```diff
@@ -32,29 +32,29 @@ const ApiClient = {
 
       request.end((err, res) => {
         if (res) {
-          const loggedData = {
+          /* const loggedData = {
```
Comment (adam-golab) [inline frontend/src/services/apiClient.js]: :bee:

File: frontend/src/store/configureStore.dev.js [hunk 8]
```diff
@@ -5,12 +5,14 @@
 import { createStore, compose, applyMiddleware } from 'redux';
 import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
 import { routerMiddleware } from 'react-router-redux';
+import thunk from 'redux-thunk';
```
Comment (adam-golab) [inline frontend/src/store/configureStore.dev.js]: What is it for? Is it necessary?

File: validations/test/schemas/loginSchema.test.js [hunk 9]
```diff
@@ -35,10 +35,11 @@ describe('Tests for input validation', () => {
   it('should return password error when too long password is provided', () => {
     const userData = {
       email: 'foo@bar.baz',
-      password: 'foobarbazfoobarbazfoobarbazfoobarbaz',
+      password: 'foobarbazfoobarbazfoobarbazfoobarbazfoobarbazfoobarbazfoo'
```
Comment (adam-golab) [inline validations/test/schemas/loginSchema.test.js]: Why not in one line?

File: frontend/src/reducers/authReducer.js (hunk 4 above)
Comment (AlejandroHerr) [inline frontend/src/reducers/authReducer.js]: It's not proper, I know. But redux-form considers that if a SubmissionErron is not thrown during submission, the submission was a success.
My opinion is that we should allow reduxPromise middleware to accept functions as REQUEST/SUCCES/FAILURE types. So we can throw and exception or (for other cases) dispatch a determined action depending on the response it got.

File: frontend/src/actions/authActions.js (hunk 3 above)
Comment (AlejandroHerr) [inline frontend/src/actions/authActions.js]: I put it so the user can see the success message (and get to know that everything went fine) while he is being redirected.

File: frontend/src/actions/thunkActions.js (hunk 6 above)
Comment (AlejandroHerr) [inline frontend/src/actions/thunkActions.js]: It's used for the redirectAfterLoginAction.

File: backend/src/user/router.spec.js (hunk 1 above)
Comment (RobertHebel) [inline backend/src/user/router.spec.js]: fixed

File: backend/src/index.js (hunk 5 above)
Comment (RobertHebel) [inline backend/src/index.js]: deleted ;)

File: frontend/src/store/configureStore.dev.js (hunk 8 above)
Comment (RobertHebel) [inline frontend/src/store/configureStore.dev.js]: deleted

File: validations/test/schemas/loginSchema.test.js (hunk 9 above)
Comment (RobertHebel) [inline validations/test/schemas/loginSchema.test.js]: fixed

File: frontend/src/services/apiClient.js (hunk 7 above)
Comment (RobertHebel) [inline frontend/src/services/apiClient.js]: fixed

**Pull request merged**